openforcefield/typing/engines/smirnoff/parameters.py

```diff
@@ -970,11 +970,17 @@ def __getitem__(self, item):
 
 
     # TODO: Override __setitem__ and __del__ to ensure we can slice by SMIRKS as well
+    # This is needed for pickling. See https://github.com/openforcefield/openforcefield/issues/411
+    # for more details.
+    # TODO: Is there a cleaner way (getstate/setstate perhaps?) to allow FFs to be
+    #       pickled?
```

Comment:
`__getstate__` and `__setstate__` would allow us full control over serialization/deserialization. I actually don't know anything about overriding `__reduce__` and can't comment on the relevant merits of this.

Reply:
In a separate PR, I'd like to really attack our serialization -- Our `to_dict` and `__init__` functions for the `ForceField` and `ParameterX` classes are, in reality, pretty close to implementations of `__getstate__` and `__setstate__`. I think overriding `__reduce__` here basically fixes the one part of `list`'s `__setstate__` that doesn't generalize to `ParameterList`s. This is a temporary but functional solution, because people are going to try and pickle ForceField objects whether we want them to or not.